yt_dlp/extractor/bilibili.py

```diff
@@ -420,14 +422,17 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, media_id)
 
         initial_state = self._search_json(r'window\.__INITIAL_STATE__\s*=', webpage, 'initial_state', media_id)
+        media_info = initial_state['mediaInfo']
         episode_list = self._download_json(
             'https://api.bilibili.com/pgc/web/season/section', media_id,
-            query={'season_id': initial_state['mediaInfo']['season_id']},
+            query={'season_id': media_info['param']['season_id']},
             note='Downloading season info')['result']['main_section']['episodes']
+        season_type = media_info['param']['season_type']
+        ep_suffix = '话' if season_type == 1 or season_type == 4 else '集'
 
         return self.playlist_result((
-            self.url_result(entry['share_url'], BiliBiliBangumiIE, entry['aid'])
-            for entry in episode_list), media_id)
+            self.url_result(entry['share_url'], BiliBiliBangumiIE, entry['aid'], (f"第{entry['title']}{ep_suffix}" if entry['title'].isnumeric() else entry['title']) + f" {entry['long_title']}")
+            for entry in episode_list), media_id, media_info['title'], media_info['evaluate'])
```

Comment:
```suggestion
            self.url_result(entry['share_url'], BiliBiliBangumiIE, entry['aid'], (
                f'第{entry["title"]}{ep_suffix}' if int_or_none(entry.get('title')
                else join_nonempty('title', 'long_title', delim=' ', from_dict=entry))
            for entry in episode_list), media_id, media_info.get('title'), media_info.get('evaluate'))
```

Make sure that the video title returned here and that returned by the single video extractor is same. i.e, `yt-dlp -O title "URL"` should give the same result with and without `--flat-playlist`

Reply:
Making it return the same title as the individual video extractor is complicated because the individual video extractor uses the text of the HTML element and here it uses the API result. the HTML element text is affected by many things and I can't guarantee that the combined text will match the HTML element text.

Reply:
The title I return is the one displayed on the page, while the title used by the video extractor is not actually visible (with `display: none`).

Reply:
If you think this one's better, you can change the video IE accordingly. Otherwise, you'd have to revert returning title here. The 2 IEs giving different titles breaks functions like `--match-filter`

Reply:
I changed the title of the video extractor so that they now return the same title.